polar-core/src/vm.rs

```diff
@@ -1772,10 +1802,10 @@ impl PolarVirtualMachine {
                 self.bind(&answer, Term::new_temporary(Value::Boolean(false)));
 
                 // append unify goal to be evaluated after external op result is returned & bound
-                self.append_goals(vec![Goal::Unify {
+                self.push_goal(Goal::Unify {
```

Comment:
Reason for this change?

Reply:
It seemed wasteful to create a vector to push a single goal, when we have a method that accepts a single goal.